components/ntp_background_images/browser/view_counter_model.cc

```diff
@@ -137,6 +137,19 @@ void ViewCounterModel::RotateBackgroundWallpaperImageIndex() {
   current_wallpaper_image_index_ %= total_image_count_;
 }
 
+void ViewCounterModel::NextBrandedImage() {
+  campaigns_current_branded_image_index_[current_campaign_index_]++;
+  if (campaigns_current_branded_image_index_[current_campaign_index_] >=
+      campaigns_total_branded_image_count_[current_campaign_index_]) {
+    campaigns_current_branded_image_index_[current_campaign_index_] = 0;
+
+    current_campaign_index_++;
+    if (current_campaign_index_ >= total_campaign_count_) {
+      current_campaign_index_ = 0;
+    }
```

Comment:
Seems like we need to reset the current branded image index after we changed the `current_campaign_index_`. Otherwise some of wallpapers won't be iterated during `ViewCounterService::GetNextBrandedWallpaperWhichMatchesConditions()` call. I think that adding of following line here should be enough:

```
campaigns_current_branded_image_index_[current_campaign_index_] = 0;
```

Reply:
Fixed